src/poliastro/twobody/orbit.py

```diff
@@ -485,51 +484,45 @@ def propagate(self, value, method=mean_motion, rtol=1e-10, **kwargs):
             parameters used in perturbation models
 
         """
-        if hasattr(value, "unit") and value.unit in ("rad", "deg"):
-            p, ecc, inc, raan, argp, _ = rv2coe(
-                self.attractor.k.to(u.km ** 3 / u.s ** 2).value,
-                self.r.to(u.km).value,
-                self.v.to(u.km / u.s).value,
-            )
-
-            # Compute time of flight for correct epoch
-            M = nu_to_M(self.nu, self.ecc)
-            new_M = nu_to_M(value, self.ecc)
-            time_of_flight = Angle(new_M - M).wrap_at(360 * u.deg) / self.n
-
-            return self.from_classical(
-                self.attractor,
-                p / (1.0 - ecc ** 2) * u.km,
-                ecc * u.one,
-                inc * u.rad,
-                raan * u.rad,
-                argp * u.rad,
-                value,
-                epoch=self.epoch + time_of_flight,
-                plane=self._plane,
-            )
+        if isinstance(value, time.Time) and not isinstance(value, time.TimeDelta):
+            time_of_flight = value - self.epoch
         else:
-            if isinstance(value, time.Time) and not isinstance(value, time.TimeDelta):
-                time_of_flight = value - self.epoch
-            else:
-                time_of_flight = time.TimeDelta(value)
+            time_of_flight = time.TimeDelta(value)
 
-            return propagate(self, time_of_flight, method=method, rtol=rtol, **kwargs)
+        return propagate(self, time_of_flight, method=method, rtol=rtol, **kwargs)
+
+    def propagate_to_anomaly(self, value):
+        """Propagates an orbit.
+
+        If value is true anomaly, propagate orbit to this anomaly and return the result.
+
+        Parameters
+        ----------
+        value : ~astropy.units.Quantity
+            True anomaly values.
+        """
+        p, ecc, inc, raan, argp, _ = rv2coe(
+            self.attractor.k.to(u.km ** 3 / u.s ** 2).value,
+            self.r.to(u.km).value,
+            self.v.to(u.km / u.s).value,
+        )
 
-    def sample(self, values=None, method=mean_motion):
+        # Compute time of flight for correct epoch
+        M = nu_to_M(self.nu, self.ecc)
+        new_M = nu_to_M(value, self.ecc)
+        time_of_flight = Angle(new_M - M).wrap_at(360 * u.deg) / self.n
+
+        return self.propagate(time_of_flight)
+
+    def sample(self, values=None):
```

Comment:
As we only have one possibility now, we can have a better default.

```suggestion
    def sample(self, values=100):
```